core/src/main/scala/com.snowplowanalytics.stream/loader/Emitter.scala

```diff
@@ -0,0 +1,190 @@
+  * http://www.apache.org/licenses/LICENSE-2.0.
+  *
+  * Unless required by applicable law or agreed to in writing,
+  * software distributed under the Apache License Version 2.0 is distributed
+  * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+  * either express or implied.
+  *
+  * See the Apache License Version 2.0 for the specific language
+  * governing permissions and limitations there under.
+  */
+
+package com.snowplowanalytics.stream.loader
+
+
+import clients.BulkSender
+import com.amazonaws.services.kinesis.connectors.UnmodifiableBuffer
+import com.amazonaws.services.kinesis.connectors.interfaces.IEmitter
+
+import scala.collection.mutable.{Buffer => SMBuffer}
+
+
+// Java
+import java.io.IOException
+import java.util.{List => List}
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Scala
+import scala.collection.mutable.ListBuffer
+
+import scala.collection.immutable.{List => SList}
+import scala.collection.JavaConverters._
+
+
+import sinks.ISink
+
+
+/**
+  * ElasticsearchEmitter class for any sort of BulkSender Extension
+  *
+  * @param bulkSender        The bulkSender Client to use
+  * @param goodSink          the configured GoodSink
+  * @param badSink           the configured BadSink
+  * @param bufferRecordLimit record limit for buffer
+  * @param bufferByteLimit   byte limit for buffer
+  */
+class Emitter(
+               bulkSender: BulkSender[EmitterJsonInput],
+               goodSink: Option[ISink],
+               badSink: ISink,
+               bufferRecordLimit: Long,
+               bufferByteLimit: Long
+             ) extends IEmitter[EmitterJsonInput] {
+
+
+
+
+  @throws[IOException]
+  def emit(buffer: UnmodifiableBuffer[EmitterJsonInput]): List[EmitterJsonInput] = {
+    attemptEmit(buffer.getRecords)
+  }
+
+
+  /**
+    * Emits good records to stdout or Elasticsearch.
+    * All records which Elasticsearch rejects and all records which failed transformation
+    * get sent to to stderr or Kinesis.
+    *
+    * @param records list containing EmitterInputs
+    * @return list of inputs which failed transformation or which Elasticsearch rejected
+    */
+  @throws[IOException]
+  def attemptEmit(records: List[EmitterJsonInput]): List[EmitterJsonInput] = {
+    if (records.asScala.isEmpty) {
+      null
+    } else {
+      val (validRecords: SList[EmitterJsonInput], invalidRecords: SList[EmitterJsonInput]) = records.asScala.toList.partition(_._2.isSuccess)
+      // Send all valid records to stdout / Elasticsearch and return those rejected by Elasticsearch
+      val rejects = goodSink match {
+        case Some(s) => {
+          validRecords.foreach(recordTuple =>
+            recordTuple.map(record => record.map(r => s.store(r.json.toString, None, true))))
+          Nil
+        }
+        case None if validRecords.isEmpty => Nil
+        case _ => emit(validRecords)
+      }
+
+      (invalidRecords ++ rejects).asJava
```

Comment:
why convert to java's list?

Reply:
`emit` needs Java list

Reply:
`emit` needs Java type for its return value and this class mainly implements KCL emitter for Java.

Reply:
ah, can you add `override` to the methods which are from `IEmitter` that will make things clearer